### PR TITLE
feat: Initial Implementation of Storager

### PR DESCRIPTION
**Pull request by abyss-w**
feat: Initial Implementation of Storager

### CODE REVIEW

**Comment by abyss-w** [comment]
UCloud US3's Go SDK has some differences from other object storages, so some APIs may not be handled correctly.

File: utils.go [hunk 1]
```diff
@@ -15,23 +44,139 @@ type Storage struct {
...
+	}
+
+	config := &us3.Config{
+		PublicKey:  ak,
+		PrivateKey: sk,
+		FileHost:   url,
+		BucketHost: "api.ucloud.cn",
+	}
+
+	srv.service, err = us3.NewFileRequest(config, nil)
```
Comment (JinnyYi) [inline utils.go:114]: Maybe we need `NewBucketRequest` here, ref: https://github.com/ufilesdk-dev/ufile-gosdk/blob/master/example/demo_bucket.go?

File: utils.go [hunk 2]
```diff
@@ -1,12 +1,41 @@
 package us3
 
 import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	uerr "github.com/ucloud/ucloud-sdk-go/ucloud/error"
+	us3 "github.com/ufilesdk-dev/ufile-gosdk"
```
Comment (Xuanwo) [inline utils.go:10]: Ops, so confusing name of `ufile` here. Strange sdk.

File: utils.go [hunk 3]
```diff
@@ -15,23 +44,139 @@ type Storage struct {
...
+}
+
+func (s *Service) newStorage(pairs ...types.Pair) (store *Storage, err error) {
+	opt, err := parsePairStorageNew(pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	store = &Storage{
+		client:  s.service,
```
Comment (JinnyYi) [inline utils.go:151]: `BucketName` should be specified for `NewFileRequest`, ref: https://github.com/ufilesdk-dev/ufile-gosdk/blob/master/request.go#L48-L49.

File: utils.go (hunk 2 above)
Comment (abyss-w) [reply utils.go:10]: The predecessor of us3 is ufile.

File: utils.go (hunk 3 above)
Comment (Xuanwo) [reply utils.go:151]:
```
//Request 创建后的 instance 不是线程安全的，如果你需要做并发的操作，请创建多个 UFileRequest。
```

It sound like we need to store a `http.Client` here and create `UFileRequest` every time we sending API.

File: utils.go (hunk 1 above)
Comment (abyss-w) [reply utils.go:114]: I think it should be `NewFileRequest`. Although they share the same model, us3 requires `NewBucketRequest` not to call file-related APIs. Like the [US3](https://github.com/ufilesdk-dev/ufile-gosdk/blob/master/request.go#:~:text=//NewBucketRequest%20%E5%88%9B%E5%BB%BA%E4%B8%80%E4%B8%AA,%E7%9B%B8%E5%85%B3%E6%96%87%E6%A1%A3%E3%80%82) comments says.

Comment (abyss-w) [reply utils.go:114]: Maybe we should separate the creation of `servicer` and `storager`, I don't think they can share a client.

Comment (JinnyYi) [reply utils.go:114]: > I think it should be `NewFileRequest`. Although they share the same model, us3 requires `NewBucketRequest` not to call file-related APIs. 

`Servicer` is the container for storage. It is used to manage the bucket here in my view and just as `NewBucketRequest` for bucket related options.

**Comment by Xuanwo** [comment]
As described in #4 , we will stop the development on us3 for now.

**Pull request closed without merging**